extensions-contrib/sqlserver-metadata-storage/src/main/java/io/druid/metadata/storage/sqlserver/SQLServerConnector.java

```diff
@@ -0,0 +1,183 @@
+package io.druid.metadata.storage.sqlserver;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.skife.jdbi.v2.Binding;
+import org.skife.jdbi.v2.ColonPrefixNamedParamStatementRewriter;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.StatementContext;
+import org.skife.jdbi.v2.tweak.HandleCallback;
+import org.skife.jdbi.v2.tweak.RewrittenStatement;
+import org.skife.jdbi.v2.tweak.StatementRewriter;
+import org.skife.jdbi.v2.util.StringMapper;
+
+import com.google.common.base.Supplier;
+import com.google.inject.Inject;
+import com.metamx.common.logger.Logger;
+
+import io.druid.metadata.MetadataStorageConnectorConfig;
+import io.druid.metadata.MetadataStorageTablesConfig;
+import io.druid.metadata.SQLMetadataConnector;
+
+@SuppressWarnings( "nls" )
+public class SQLServerConnector extends SQLMetadataConnector
+{
+  private static final Logger log = new Logger(SQLServerConnector.class);
+
+  private static final String PAYLOAD_TYPE = "VARBINARY(MAX)";
+  private static final String SERIAL_TYPE = "[int] IDENTITY (1, 1)";
+
+  public static final int DEFAULT_STREAMING_RESULT_SIZE = 100;
```

Comment:
Will add code comments in relation to this.  On a closer look, I just noticed that I should have used a [bigint] instead of a [int] for the SERIAL_TYPE.